cmake error in clang32 for mingw-w64-dcmtk

```
  -- Looking for socket in socket - not found
  CMake Error at CMake/GenerateDCMTKConfigure.cmake:1433 (message):
    Exit code 0xc0000139
  -- Configuring incomplete, errors occurred!
  
  See also "C:/_/mingw-w64-dcmtk/src/build-CLANG32/CMakeFiles/CMakeOutput.log".
  Call Stack (most recent call first):
    CMakeLists.txt:52 (INSPECT_FUNDAMENTAL_ARITHMETIC_TYPES)
  
  
  See also "C:/_/mingw-w64-dcmtk/src/build-CLANG32/CMakeFiles/CMakeError.log".
  ==> ERROR: A failure occurred in build().
```

This only seems to happen in clang32.  https://github.com/jeremyd2019/MINGW-packages/actions/runs/3448053806/jobs/5754703669

--- a/mingw-w64-dcmtk/002-properly-detect-canonical-host.patch
+++ b/mingw-w64-dcmtk/002-properly-detect-canonical-host.patch
@@ -1,0 +1,12 @@
+diff -ur dcmtk-3.6.7-orig/CMake/GenerateDCMTKConfigure.cmake dcmtk-3.6.7/CMake/GenerateDCMTKConfigure.cmake
+--- dcmtk-3.6.7-orig/CMake/GenerateDCMTKConfigure.cmake	2022-04-28 21:47:25.000000000 +0800
++++ dcmtk-3.6.7/CMake/GenerateDCMTKConfigure.cmake	2022-11-14 20:52:21.642075900 +0800
+@@ -120,7 +120,7 @@
+ 
+ # Windows being windows, it lies about its processor type to 32 bit binaries
+ set(SYSTEM_PROCESSOR "$ENV{PROCESSOR_ARCHITEW6432}")
+-if(NOT SYSTEM_PROCESSOR)
++if(NOT SYSTEM_PROCESSOR OR SYSTEM_PROCESSOR STREQUAL "AMD64")
+   if(WIN32 AND NOT CYGWIN)
+     if(CMAKE_GENERATOR_PLATFORM)
+       set(SYSTEM_PROCESSOR "${CMAKE_GENERATOR_PLATFORM}")

--- a/mingw-w64-dcmtk/PKGBUILD
+++ b/mingw-w64-dcmtk/PKGBUILD
@@ -4,12 +4,12 @@ _realname=dcmtk
 pkgbase=mingw-w64-${_realname}
 pkgname=("${MINGW_PACKAGE_PREFIX}-${_realname}")
 pkgver=3.6.7
-pkgrel=2
+pkgrel=3
 pkgdesc="A collection of libraries and applications implementing large parts of the DICOM standard (mingw-w64)"
 arch=('any')
-mingw_arch=('mingw32' 'mingw64' 'ucrt64' 'clang64')
+mingw_arch=('mingw32' 'mingw64' 'ucrt64' 'clang64' 'clang32')
 url='http://dicom.offis.de/dcmtk'
-license=('custom')
+license=('spdx:BSD-3-Clause')
 depends=("${MINGW_PACKAGE_PREFIX}-icu"
          "${MINGW_PACKAGE_PREFIX}-libiconv"
          "${MINGW_PACKAGE_PREFIX}-libpng"
@@ -21,20 +21,37 @@ depends=("${MINGW_PACKAGE_PREFIX}-icu"
          "${MINGW_PACKAGE_PREFIX}-zlib")
 makedepends=("${MINGW_PACKAGE_PREFIX}-cmake"
              "${MINGW_PACKAGE_PREFIX}-ninja"
+             "${MINGW_PACKAGE_PREFIX}-autotools"
              "${MINGW_PACKAGE_PREFIX}-cc")
 source=("https://dicom.offis.de/download/dcmtk/dcmtk${pkgver//./}/${_realname}-${pkgver}.tar.gz"
-        "001-fix-cmake-config-install-destination.patch")
+        "001-fix-cmake-config-install-destination.patch"
+        "002-properly-detect-canonical-host.patch")
 sha256sums=('7c58298e3e8d60232ee6fc8408cfadd14463cc11a3c4ca4c59af5988c7e9710a'
-            'ff124b6f26c5a5982268597c6285e2e320a248a090046b45618e79cb8a6fc099')
+            'ff124b6f26c5a5982268597c6285e2e320a248a090046b45618e79cb8a6fc099'
+            'b9cd641267f9aeb055608f045cc91ad5b99b071897aa849ed3282aa03b5ae925')
 
 prepare() {
   cd "${srcdir}"/${_realname}-${pkgver}
   patch -p1 -i "${srcdir}"/001-fix-cmake-config-install-destination.patch
+  patch -p1 -i "${srcdir}"/002-properly-detect-canonical-host.patch
+
+  cd config
+  ./rootconf
 }
 
 build() {
   [[ -d "${srcdir}"/build-${MSYSTEM} ]] && rm -rf "${srcdir}"/build-${MSYSTEM}
-  mkdir -p "${srcdir}"/build-${MSYSTEM} && cd "${srcdir}"/build-${MSYSTEM}
+  mkdir -p "${srcdir}"/build-${MSYSTEM}
+
+  # this is just to get arith.h
+  cd "${srcdir}"/${_realname}-${pkgver}
+
+  ./configure --ignore-deprecation
+  make config-tests-all
+  install -D config/include/dcmtk/config/arith.h "${srcdir}"/build-${MSYSTEM}/config/include/dcmtk/config/arith.h
+
+  # continue the build using cmake
+  cd "${srcdir}"/build-${MSYSTEM}
 
   declare -a extra_config
   if check_option "debug" "n"; then

--- a/mingw-w64-openimageio/PKGBUILD
+++ b/mingw-w64-openimageio/PKGBUILD
@@ -4,7 +4,7 @@ _realname=openimageio
 pkgbase=mingw-w64-${_realname}
 pkgname="${MINGW_PACKAGE_PREFIX}-${_realname}"
 pkgver=2.4.5.0
-pkgrel=2
+pkgrel=3
 pkgdesc="A library for reading and writing images, including classes, utilities, and applications (mingw-w64)"
 arch=('any')
 mingw_arch=('mingw32' 'mingw64' 'ucrt64' 'clang64' 'clang32')
@@ -12,7 +12,7 @@ url="https://www.openimageio.org/"
 license=('spdx:BSD-3-Clause')
 depends=("${MINGW_PACKAGE_PREFIX}-boost"
          "${MINGW_PACKAGE_PREFIX}-bzip2"
-         $([[ "${MSYSTEM}" == "CLANG32" ]] || echo "${MINGW_PACKAGE_PREFIX}-dcmtk")
+         "${MINGW_PACKAGE_PREFIX}-dcmtk"
          "${MINGW_PACKAGE_PREFIX}-ffmpeg"
          "${MINGW_PACKAGE_PREFIX}-freetype"
          "${MINGW_PACKAGE_PREFIX}-fmt"

